src/test/java/gov/usgs/wma/waterdata/groundwater/RdbWriterTest.java

```diff
@@ -229,28 +294,78 @@ public Writer append(CharSequence csq) throws IOException {
 	// These methods used to be located in the superclass.
 	// They were added only for testing and thus moved here.
 	class MockRdbWriter extends RdbWriter {
+
+		String headerRaw;
+		ArrayList<String> dataRows = new ArrayList();
+
 		public MockRdbWriter(Writer destination) {
 			super(destination);
 		}
 
-		protected MockRdbWriter flush() {
+		private void flush() {
 			try {
 				rdb.flush();
 			} catch (IOException e) {
-				// do not care about flush exceptions
+				e.printStackTrace(); //ignore
 			}
+		}
+		@Override
+		public RdbWriter writeHeader() {
+			super.writeHeader();
+			flush();
+			headerRaw = out.toString();
 			return this;
 		}
```

Comment:
This method no longer needs to be explicitly called from the test, so I internalized it.